data/binding/convert.go

```diff
@@ -258,7 +258,7 @@ func (s *toStringFrom[T]) Set(str string) error {
 		return err
 	}
 
-	queueItem(s.DataChanged)
+	s.DataChanged()
```

Comment:
Does this need to be in fyne.Do?

Reply:
Like I wrote in the description, `s.DataChanged()` just runs `.trigger()` which queues the operation into main. Having a second fyne.Do here is not needed.

Reply:
I'm not totally sure I agree here - the `DataChanged` is part of the public API and must also be called on the correct thread.